src/views/Gantt.vue

```diff
@@ -0,0 +1,267 @@
+        </v-col>
+        <v-col
+          cols="12"
+          md="4"
+          class="pl-md-2 mb-2 mb-md-0"
+        >
+          <v-select
+            id="c-gantt-filter-job-platforms"
+            :items="platformOptions"
+            prefix="Platform:"
+            v-model="jobsFilter.platformOption"
+          />
+        </v-col>
+      </v-row>
+      <v-row no-gutters>
+        <v-col
+          cols="12"
+          md="4"
+          class="pr-md-2 mb-2 mb-md-0"
+        >
+          <v-select
+            id="c-gantt-tasks-per-page"
+            :items="$options.taskChoices"
+            prefix="Tasks Per Page"
+            v-model="tasksPerPage"/>
+        </v-col>
+      </v-row>
+      <div
+        id="gantt-toolbar"
+        class="d-flex align-center flex-wrap my-2 col-gap-2 row-gap-4">
+
+        <!-- Toolbar -->
+        <v-defaults-provider>
+
+          <!-- Box plot sort input teleports here -->
+        </v-defaults-provider>
+      </div>
+      <GanttChart
+        :jobs="filteredJobs"
+        :timing-option="timingOption"
+        :tasks-per-page="tasksPerPage"
+      />
+    </v-container>
+  </div>
+</template>
+
+<script>
+import {
+  debounce
+} from 'lodash'
+import gql from 'graphql-tag'
+import { getPageTitle } from '@/utils/index'
+import graphqlMixin from '@/mixins/graphql'
+import GanttChart from '@/components/cylc/gantt/GanttChart.vue'
+import {
+  matchTasks,
+  platformOptions
+} from '@/components/cylc/analysis/filter'
+/** List of fields to request for each job */
+const jobFields = [
+  'name',
+  'id',
+  'submittedTime',
+  'startedTime',
+  'finishedTime',
+  'platform'
+]
+/** The query which retrieves historical Job timing statistics */
+const QUERY = gql`
+query ganttQuery ($workflows: [ID]) {
+  jobs(live: false, workflows: $workflows) {
+    ${jobFields.join('\n')}
+  }
+}
+`
+/** The callback which gets called when data comes in from the query */
+export class GanttCallback {
+  /**
+   * @param {Object[]} jobs
+   */
+  constructor (jobs) {
+    this.jobs = jobs
+  }
+  /**
+   * Add jobs contained in data to this.jobs
+   */
+
+  add (data) {
+    const taskNameList = Array.from(
+      new Set(data.jobs.map((job) => job.name))
+    )
+    this.uniqueTasks = Array.from(
+      new Set(data.jobs.map((job) => job.name))
+    )
+    const sortedData = Object.fromEntries(taskNameList.map(key => [key, []]))
+    for (let i = 0; i < data.jobs.length; i++) {
+      sortedData[data.jobs[i].name].push(data.jobs[i])
+    }
+    this.jobs.push(sortedData)
+  }
+  // called when new objects are added
+  // NOTE: we manually call this to add items which come through on the query
+
+  onAdded (added, store, errors) {
+    this.add(added)
+  }
+
+  // called when existing objects are updated
+  onUpdated (updated, store, errors) {
+    this.add(updated)
+  }
+
+  // other hooks we don't need but must declare (for now)
+  before () {}
+  after () {}
+  onPruned () {}
+  commit () {}
+  tearDown () {}
```

Comment:
MB. If we inherit from the parent class as Ronnie suggested, then we don't have to define these blank methods.